Build error on 32 bit RaspberryPI

<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my issue will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated.

## OS / platform the server is running (if known)

<!-- Windows10 / Ubuntu / Mac / ARM etc --> raspberry pi os 32 bit

## Branch affected by issue <!-- Change to the branch the issue exists on (if relevant) -->

`base`

## Steps to reproduce

<!-- Add steps to reproduce here -->

## Expected behavior

<!-- Add expected behaviour here -->

----

[ 29%] Built target xi_search
/usr/bin/ld: /tmp/xi_world.2dkwgt.ltrans7.ltrans.o: in function `HTTPServer::LockingUpdate()':
/usr/include/c++/10/atomic:258: undefined reference to `__atomic_load_8'
/usr/bin/ld: /usr/include/c++/10/atomic:247: undefined reference to `__atomic_store_8'
collect2: error: ld returned 1 exit status
make[2]: *** [src/world/CMakeFiles/xi_world.dir/build.make:508: ../xi_world] Error 1
make[1]: *** [CMakeFiles/Makefile2:1082: src/world/CMakeFiles/xi_world.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

> which raspi is this? what's your operating system?
> 
> Ubuntu 22.04 is tested to work on Raspi4 and the compute module. Debian/Raspian should work with new enough libs and GCC, but I don't have a great test bed for it.

Its a fully up to date Raspberry pi OS (Debian 11)... it used to build fine.

> Please post your full logs, you're missing `-latomic` flags while compiling/linking
> 
> Looks similar to this: https://gitlab.kitware.com/cmake/cmake/-/issues/21174
```
-- The C compiler identification is GNU 10.2.1
-- The CXX compiler identification is GNU 10.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Downloading CPM.cmake to /home/pi/lsb/server/build/cmake/CPM_0.34.0.cmake
-- CMAKE_SOURCE_DIR: /home/pi/lsb/server
-- CMAKE_SIZEOF_VOID_P == 4: 32-bit build
-- Setting build type to 'RelWithDebInfo' as none was specified.
-- ENABLE_FAST_MATH: ON
-- ENABLE_VALGRIND: OFF
-- Found Git: /usr/bin/git (found version "2.30.2") 
-- GIT_SHA1: 4f31e40b9b-dirty
-- GIT_BRANCH: base
-- GIT_DATE: Fri Feb 17 16:35:02 2023
-- GIT_COMMIT_SUBJECT: wiill this build
-- CMAKE_VERSION: 3.18.4
-- CMAKE_C_COMPILER: /usr/bin/cc
-- CMAKE_CXX_COMPILER: /usr/bin/c++
-- CMAKE_CXX_COMPILER_ID: GNU
-- CMAKE_CXX_COMPILER_VERSION: 10.2.1
-- CMAKE_GENERATOR: Unix Makefiles
-- CMAKE_GENERATOR_PLATFORM: 
-- CMAKE_BUILD_TYPE: RelWithDebInfo
-- CMAKE_CXX_FLAGS:  
-- CMAKE_CXX_FLAGS_RELWITHDEBINFO: -O2 -g -DNDEBUG
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Found MariaDB: /usr/lib/arm-linux-gnueabihf/libmariadb.so  
-- MARIADB_FOUND: TRUE
-- MARIADB_LIBRARY: /usr/lib/arm-linux-gnueabihf/libmariadb.so
-- MARIADB_INCLUDE_DIR: /home/pi/lsb/server/ext/mariadb/include/
-- Found ZeroMQ: /usr/lib/arm-linux-gnueabihf/libzmq.so  
-- ZeroMQ_FOUND: TRUE
-- ZeroMQ_LIBRARY: /usr/lib/arm-linux-gnueabihf/libzmq.so
-- ZeroMQ_INCLUDE_DIR: /home/pi/lsb/server/ext/zmq/include/zmq/
-- Found LuaJIT: /usr/lib/arm-linux-gnueabihf/libluajit-5.1.so  
-- LuaJIT_FOUND: TRUE
-- LuaJIT_LIBRARY: /usr/lib/arm-linux-gnueabihf/libluajit-5.1.so
-- LuaJIT_INCLUDE_DIR: /home/pi/lsb/server/ext/luajit/include/
-- Found Binutils: /usr/lib/arm-linux-gnueabihf/libbfd.so  
-- Binutils_FOUND: TRUE
-- Binutils_LIBRARY: /usr/lib/arm-linux-gnueabihf/libbfd.so
-- TRACY_ENABLE: OFF
-- CPM: adding package spdlog@1.9.2 (v1.9.2)
-- Build spdlog: 1.9.2
-- Build type: RelWithDebInfo
-- CPM: adding package recastnavigation@0 (67c36bda750f36b2b4152db57a23170ed7a0683c)
-- CPM: adding package fast_obj@0 (85778da5fc320b7e52885f8e869edc079695cc79)
-- CPM: adding package argparse@0 (f901d9edb47d45a287ca44d5bab48540c1fcdcce)
-- CPM: adding package task_system@0 (241acb039152b646589fcbbcfb4d8d8e5bddc815)
-- CPM: adding package efsw@0 (74ca09bff89bc8de1f7b8bf3faaa6275ce23b4c5)
-- CPM: adding package jthread-lite@0 (5332bbd46dcba5f028a844795cc0931e9f2ffdf4)
-- Project 'jthread_lite', package 'jthread-lite' version: '0.1.0'
-- CPM: adding package cpp-httplib@0.11.3 (v0.11.3)
-- CPM: adding package json@3.11.2 (v3.11.2)
-- Using the multi-header code from /home/pi/lsb/server/build/_deps/json-src/include/
-- ENABLE_CLANG_TIDY: OFF
-- ENABLE_CLANG_TIDY_AUTO_FIX: OFF
-- Configuring src/common/version.cpp
-- Setting output directory for xi_connect to /home/pi/lsb/server
-- Adding module files to build: 
-- Setting output directory for xi_map to /home/pi/lsb/server
-- Setting output directory for xi_search to /home/pi/lsb/server
-- Setting output directory for xi_world to /home/pi/lsb/server
-- Configuring done
-- Generating done
-- Build files have been written to: /home/pi/lsb/server/build
pi@raspberrypi:~/lsb/server/build $ make -j $(nproc)
Scanning dependencies of target Detour
Scanning dependencies of target spdlog
Scanning dependencies of target fast_obj_lib
Scanning dependencies of target Recast
[  0%] Building C object _deps/fast_obj-build/CMakeFiles/fast_obj_lib.dir/fast_obj.c.o
[  0%] Building CXX object _deps/recastnavigation-build/Detour/CMakeFiles/Detour.dir/Source/DetourAlloc.cpp.o
[  0%] Building CXX object _deps/recastnavigation-build/Recast/CMakeFiles/Recast.dir/Source/Recast.cpp.o
[  0%] Building CXX object _deps/spdlog-build/CMakeFiles/spdlog.dir/src/spdlog.cpp.o
[  0%] Building CXX object _deps/recastnavigation-build/Detour/CMakeFiles/Detour.dir/Source/DetourAssert.cpp.o
[  0%] Building CXX object _deps/recastnavigation-build/Detour/CMakeFiles/Detour.dir/Source/DetourCommon.cpp.o
[  0%] Linking C static library libfast_obj_lib.a
[  0%] Building CXX object _deps/recastnavigation-build/Recast/CMakeFiles/Recast.dir/Source/RecastAlloc.cpp.o
[  0%] Built target fast_obj_lib
Scanning dependencies of target efsw
[  1%] Building CXX object _deps/recastnavigation-build/Detour/CMakeFiles/Detour.dir/Source/DetourNavMesh.cpp.o
[  2%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/Debug.cpp.o
[  2%] Building CXX object _deps/recastnavigation-build/Recast/CMakeFiles/Recast.dir/Source/RecastArea.cpp.o
[  3%] Building CXX object _deps/recastnavigation-build/Recast/CMakeFiles/Recast.dir/Source/RecastAssert.cpp.o
[  3%] Building CXX object _deps/recastnavigation-build/Recast/CMakeFiles/Recast.dir/Source/RecastContour.cpp.o
[  3%] Building CXX object _deps/recastnavigation-build/Detour/CMakeFiles/Detour.dir/Source/DetourNavMeshBuilder.cpp.o
[  3%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/DirectorySnapshot.cpp.o
[  3%] Building CXX object _deps/recastnavigation-build/Recast/CMakeFiles/Recast.dir/Source/RecastFilter.cpp.o
[  3%] Building CXX object _deps/recastnavigation-build/Detour/CMakeFiles/Detour.dir/Source/DetourNavMeshQuery.cpp.o
[  3%] Building CXX object _deps/recastnavigation-build/Recast/CMakeFiles/Recast.dir/Source/RecastLayers.cpp.o
[  3%] Building CXX object _deps/recastnavigation-build/Recast/CMakeFiles/Recast.dir/Source/RecastMesh.cpp.o
[  3%] Building CXX object _deps/recastnavigation-build/Detour/CMakeFiles/Detour.dir/Source/DetourNode.cpp.o
[  4%] Building CXX object _deps/recastnavigation-build/Recast/CMakeFiles/Recast.dir/Source/RecastMeshDetail.cpp.o
[  4%] Linking CXX static library libDetour.a
[  4%] Built target Detour
[  4%] Building CXX object _deps/recastnavigation-build/Recast/CMakeFiles/Recast.dir/Source/RecastRasterization.cpp.o
[  4%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/DirectorySnapshotDiff.cpp.o
[  4%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/DirWatcherGeneric.cpp.o
[  4%] Building CXX object _deps/recastnavigation-build/Recast/CMakeFiles/Recast.dir/Source/RecastRegion.cpp.o
[  4%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/FileInfo.cpp.o
[  4%] Linking CXX static library libRecast.a
[  4%] Built target Recast
Scanning dependencies of target DetourTileCache
[  4%] Building CXX object _deps/recastnavigation-build/DetourTileCache/CMakeFiles/DetourTileCache.dir/Source/DetourTileCache.cpp.o
[  4%] Building CXX object _deps/recastnavigation-build/DetourTileCache/CMakeFiles/DetourTileCache.dir/Source/DetourTileCacheBuilder.cpp.o
Scanning dependencies of target DetourCrowd
[  5%] Building CXX object _deps/recastnavigation-build/DetourCrowd/CMakeFiles/DetourCrowd.dir/Source/DetourCrowd.cpp.o
[  6%] Linking CXX static library libDetourTileCache.a
[  7%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/FileSystem.cpp.o
[  7%] Built target DetourTileCache
[  7%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/FileWatcher.cpp.o
[  7%] Building CXX object _deps/recastnavigation-build/DetourCrowd/CMakeFiles/DetourCrowd.dir/Source/DetourLocalBoundary.cpp.o
[  7%] Building CXX object _deps/recastnavigation-build/DetourCrowd/CMakeFiles/DetourCrowd.dir/Source/DetourObstacleAvoidance.cpp.o
[  7%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/FileWatcherCWrapper.cpp.o
[  7%] Building CXX object _deps/recastnavigation-build/DetourCrowd/CMakeFiles/DetourCrowd.dir/Source/DetourPathCorridor.cpp.o
[  8%] Building CXX object _deps/recastnavigation-build/DetourCrowd/CMakeFiles/DetourCrowd.dir/Source/DetourPathQueue.cpp.o
[  8%] Building CXX object _deps/recastnavigation-build/DetourCrowd/CMakeFiles/DetourCrowd.dir/Source/DetourProximityGrid.cpp.o
[  8%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/FileWatcherGeneric.cpp.o
[  8%] Linking CXX static library libDetourCrowd.a
[  8%] Built target DetourCrowd
[  9%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/FileWatcherImpl.cpp.o
[  9%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/Log.cpp.o
[  9%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/Mutex.cpp.o
[  9%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/String.cpp.o
Scanning dependencies of target DebugUtils
[  9%] Building CXX object _deps/recastnavigation-build/DebugUtils/CMakeFiles/DebugUtils.dir/Source/DebugDraw.cpp.o
[ 10%] Building CXX object _deps/spdlog-build/CMakeFiles/spdlog.dir/src/stdout_sinks.cpp.o
[ 10%] Building CXX object _deps/recastnavigation-build/DebugUtils/CMakeFiles/DebugUtils.dir/Source/DetourDebugDraw.cpp.o
[ 10%] Building CXX object _deps/recastnavigation-build/DebugUtils/CMakeFiles/DebugUtils.dir/Source/RecastDebugDraw.cpp.o
[ 10%] Building CXX object _deps/recastnavigation-build/DebugUtils/CMakeFiles/DebugUtils.dir/Source/RecastDump.cpp.o
[ 11%] Linking CXX static library libDebugUtils.a
[ 11%] Built target DebugUtils
[ 12%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/System.cpp.o
[ 12%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/Thread.cpp.o
[ 12%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/Watcher.cpp.o
[ 12%] Building CXX object _deps/spdlog-build/CMakeFiles/spdlog.dir/src/color_sinks.cpp.o
[ 12%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/WatcherGeneric.cpp.o
[ 12%] Building CXX object _deps/spdlog-build/CMakeFiles/spdlog.dir/src/file_sinks.cpp.o
[ 12%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/platform/posix/FileSystemImpl.cpp.o
[ 13%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/platform/posix/MutexImpl.cpp.o
[ 13%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/platform/posix/SystemImpl.cpp.o
[ 13%] Building CXX object _deps/spdlog-build/CMakeFiles/spdlog.dir/src/async.cpp.o
[ 14%] Building CXX object _deps/spdlog-build/CMakeFiles/spdlog.dir/src/cfg.cpp.o
[ 14%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/platform/posix/ThreadImpl.cpp.o
[ 14%] Building CXX object _deps/spdlog-build/CMakeFiles/spdlog.dir/src/fmt.cpp.o
[ 14%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/FileWatcherInotify.cpp.o
In file included from /home/pi/lsb/server/build/_deps/efsw-src/src/efsw/FileWatcherInotify.cpp:13:
/home/pi/lsb/server/build/_deps/efsw-src/src/efsw/inotify-nosys.h:68: warning: "__NR_inotify_init" redefined
   68 | # define __NR_inotify_init (__NR_SYSCALL_BASE+316)
      | 
In file included from /usr/include/arm-linux-gnueabihf/asm/unistd.h:22,
                 from /usr/include/arm-linux-gnueabihf/sys/syscall.h:24,
                 from /home/pi/lsb/server/build/_deps/efsw-src/src/efsw/inotify-nosys.h:5,
                 from /home/pi/lsb/server/build/_deps/efsw-src/src/efsw/FileWatcherInotify.cpp:13:
/usr/include/arm-linux-gnueabihf/asm/unistd-eabi.h:275: note: this is the location of the previous definition
  275 | #define __NR_inotify_init (__NR_SYSCALL_BASE + 316)
      | 
In file included from /home/pi/lsb/server/build/_deps/efsw-src/src/efsw/FileWatcherInotify.cpp:13:
/home/pi/lsb/server/build/_deps/efsw-src/src/efsw/inotify-nosys.h:69: warning: "__NR_inotify_add_watch" redefined
   69 | # define __NR_inotify_add_watch (__NR_SYSCALL_BASE+317)
      | 
In file included from /usr/include/arm-linux-gnueabihf/asm/unistd.h:22,
                 from /usr/include/arm-linux-gnueabihf/sys/syscall.h:24,
                 from /home/pi/lsb/server/build/_deps/efsw-src/src/efsw/inotify-nosys.h:5,
                 from /home/pi/lsb/server/build/_deps/efsw-src/src/efsw/FileWatcherInotify.cpp:13:
/usr/include/arm-linux-gnueabihf/asm/unistd-eabi.h:276: note: this is the location of the previous definition
  276 | #define __NR_inotify_add_watch (__NR_SYSCALL_BASE + 317)
      | 
In file included from /home/pi/lsb/server/build/_deps/efsw-src/src/efsw/FileWatcherInotify.cpp:13:
/home/pi/lsb/server/build/_deps/efsw-src/src/efsw/inotify-nosys.h:70: warning: "__NR_inotify_rm_watch" redefined
   70 | # define __NR_inotify_rm_watch (__NR_SYSCALL_BASE+318)
      | 
In file included from /usr/include/arm-linux-gnueabihf/asm/unistd.h:22,
                 from /usr/include/arm-linux-gnueabihf/sys/syscall.h:24,
                 from /home/pi/lsb/server/build/_deps/efsw-src/src/efsw/inotify-nosys.h:5,
                 from /home/pi/lsb/server/build/_deps/efsw-src/src/efsw/FileWatcherInotify.cpp:13:
/usr/include/arm-linux-gnueabihf/asm/unistd-eabi.h:277: note: this is the location of the previous definition
  277 | #define __NR_inotify_rm_watch (__NR_SYSCALL_BASE + 318)
      | 
[ 15%] Building CXX object _deps/efsw-build/CMakeFiles/efsw.dir/src/efsw/WatcherInotify.cpp.o
[ 15%] Linking CXX static library libefsw.a
[ 15%] Built target efsw
[ 15%] Linking CXX static library libspdlog.a
[ 15%] Built target spdlog
Scanning dependencies of target xi_connect
Scanning dependencies of target xi_world
Scanning dependencies of target xi_search
Scanning dependencies of target xi_map
[ 15%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/application.cpp.o
[ 15%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/application.cpp.o
[ 15%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/application.cpp.o
[ 15%] Building CXX object src/map/CMakeFiles/xi_map.dir/cmake_pch.hxx.gch
[ 17%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/async.cpp.o
[ 17%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/async.cpp.o
[ 17%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/async.cpp.o
[ 18%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/application.cpp.o
[ 18%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/blowfish.cpp.o
[ 18%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/blowfish.cpp.o
[ 19%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/blowfish.cpp.o
[ 19%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/console_service.cpp.o
[ 19%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/console_service.cpp.o
[ 19%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/console_service.cpp.o
[ 19%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/async.cpp.o
[ 19%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/blowfish.cpp.o
[ 19%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/filewatcher.cpp.o
[ 19%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/filewatcher.cpp.o
[ 19%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/filewatcher.cpp.o
[ 19%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/console_service.cpp.o
[ 19%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/filewatcher.cpp.o
[ 19%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/kernel.cpp.o
[ 20%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/logging.cpp.o
[ 21%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/kernel.cpp.o
[ 22%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/kernel.cpp.o
[ 22%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/lua.cpp.o
[ 22%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/logging.cpp.o
[ 23%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/logging.cpp.o
[ 23%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/logging.cpp.o
[ 23%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/lua.cpp.o
[ 23%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/md52.cpp.o
[ 23%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/lua.cpp.o
[ 23%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/lua.cpp.o
[ 23%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/settings.cpp.o
[ 23%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/md52.cpp.o
[ 24%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/settings.cpp.o
[ 24%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/md52.cpp.o
[ 24%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/md52.cpp.o
[ 24%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/settings.cpp.o
[ 24%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/settings.cpp.o
[ 24%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/sql_prepared_stmt.cpp.o
[ 24%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/socket.cpp.o
[ 25%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/sql.cpp.o
[ 25%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/sql_prepared_stmt.cpp.o
[ 25%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/sql.cpp.o
[ 25%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/socket.cpp.o
[ 26%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/socket.cpp.o
[ 26%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/taskmgr.cpp.o
[ 26%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/taskmgr.cpp.o
[ 27%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/timer.cpp.o
[ 27%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/utils.cpp.o
[ 27%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/timer.cpp.o
[ 27%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/sql_prepared_stmt.cpp.o
[ 28%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/sql_prepared_stmt.cpp.o
[ 28%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/utils.cpp.o
[ 28%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/sql.cpp.o
[ 28%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/sql.cpp.o
[ 28%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/vana_time.cpp.o
[ 28%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/version.cpp.o
[ 29%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/watchdog.cpp.o
[ 29%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/zlib.cpp.o
[ 29%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/taskmgr.cpp.o
[ 29%] Building CXX object src/map/CMakeFiles/xi_map.dir/__/common/debug_linux.cpp.o
[ 29%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/taskmgr.cpp.o
[ 30%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/vana_time.cpp.o
[ 30%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/controllers/automaton_controller.cpp.o
[ 30%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/version.cpp.o
[ 30%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/watchdog.cpp.o
[ 31%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/timer.cpp.o
[ 31%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/timer.cpp.o
[ 31%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/zlib.cpp.o
[ 31%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/utils.cpp.o
[ 32%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/utils.cpp.o
[ 32%] Building CXX object src/world/CMakeFiles/xi_world.dir/__/common/debug_linux.cpp.o
[ 32%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/controllers/controller.cpp.o
[ 32%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/vana_time.cpp.o
[ 33%] Building CXX object src/world/CMakeFiles/xi_world.dir/besieged_system.cpp.o
[ 33%] Building CXX object src/world/CMakeFiles/xi_world.dir/campaign_system.cpp.o
[ 33%] Building CXX object src/world/CMakeFiles/xi_world.dir/colonization_system.cpp.o
[ 33%] Building CXX object src/world/CMakeFiles/xi_world.dir/conquest_system.cpp.o
[ 34%] Building CXX object src/world/CMakeFiles/xi_world.dir/http_server.cpp.o
[ 34%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/vana_time.cpp.o
[ 34%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/version.cpp.o
[ 35%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/watchdog.cpp.o
[ 36%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/controllers/mob_controller.cpp.o
[ 36%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/version.cpp.o
[ 36%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/watchdog.cpp.o
[ 36%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/zlib.cpp.o
[ 37%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/zlib.cpp.o
[ 37%] Building CXX object src/search/CMakeFiles/xi_search.dir/__/common/debug_linux.cpp.o
[ 37%] Building CXX object src/login/CMakeFiles/xi_connect.dir/__/common/debug_linux.cpp.o
[ 37%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/controllers/pet_controller.cpp.o
[ 37%] Building CXX object src/search/CMakeFiles/xi_search.dir/packets/auction_history.cpp.o
[ 37%] Building CXX object src/login/CMakeFiles/xi_connect.dir/lobby.cpp.o
[ 37%] Building CXX object src/search/CMakeFiles/xi_search.dir/packets/auction_list.cpp.o
[ 38%] Building CXX object src/search/CMakeFiles/xi_search.dir/packets/linkshell_list.cpp.o
[ 38%] Building CXX object src/world/CMakeFiles/xi_world.dir/main.cpp.o
[ 38%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/controllers/player_charm_controller.cpp.o
[ 38%] Building CXX object src/search/CMakeFiles/xi_search.dir/packets/party_list.cpp.o
[ 38%] Building CXX object src/search/CMakeFiles/xi_search.dir/packets/search_comment.cpp.o
[ 38%] Building CXX object src/login/CMakeFiles/xi_connect.dir/login_auth.cpp.o
[ 38%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/controllers/player_controller.cpp.o
[ 38%] Building CXX object src/search/CMakeFiles/xi_search.dir/packets/search_list.cpp.o
[ 38%] Building CXX object src/world/CMakeFiles/xi_world.dir/message_server.cpp.o
[ 39%] Building CXX object src/search/CMakeFiles/xi_search.dir/data_loader.cpp.o
[ 39%] Building CXX object src/login/CMakeFiles/xi_connect.dir/login_conf.cpp.o
[ 40%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/controllers/trust_controller.cpp.o
[ 40%] Building CXX object src/world/CMakeFiles/xi_world.dir/world_server.cpp.o
[ 40%] Building CXX object src/search/CMakeFiles/xi_search.dir/search.cpp.o
[ 41%] Building CXX object src/login/CMakeFiles/xi_connect.dir/login_session.cpp.o
[ 41%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/helpers/action_queue.cpp.o
[ 41%] Building CXX object src/search/CMakeFiles/xi_search.dir/tcp_request.cpp.o
[ 41%] Building CXX object src/login/CMakeFiles/xi_connect.dir/login.cpp.o
[ 42%] Linking CXX executable ../../../xi_world
[ 42%] Linking CXX executable ../../../xi_search
[ 42%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/helpers/event_handler.cpp.o
[ 42%] Linking CXX executable ../../../xi_connect
[ 42%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/helpers/gambits_container.cpp.o
[ 42%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/helpers/pathfind.cpp.o
[ 43%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/helpers/targetfind.cpp.o
[ 43%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/states/ability_state.cpp.o
[ 43%] Built target xi_search
[ 43%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/states/attack_state.cpp.o
[ 43%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/states/death_state.cpp.o
[ 44%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/states/despawn_state.cpp.o
/usr/bin/ld: /tmp/xi_world.yfNqsG.ltrans7.ltrans.o: in function `HTTPServer::LockingUpdate()':
/usr/include/c++/10/atomic:258: undefined reference to `__atomic_load_8'
/usr/bin/ld: /usr/include/c++/10/atomic:247: undefined reference to `__atomic_store_8'
[ 44%] Built target xi_connect
[ 44%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/states/inactive_state.cpp.o
collect2: error: ld returned 1 exit status
make[2]: *** [src/world/CMakeFiles/xi_world.dir/build.make:508: ../xi_world] Error 1
make[1]: *** [CMakeFiles/Makefile2:1082: src/world/CMakeFiles/xi_world.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 44%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/states/item_state.cpp.o
[ 44%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/states/magic_state.cpp.o
[ 45%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/states/mobskill_state.cpp.o
[ 45%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/states/petskill_state.cpp.o
[ 45%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/states/raise_state.cpp.o
[ 45%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/states/range_state.cpp.o
[ 45%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/states/respawn_state.cpp.o
[ 46%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/states/state.cpp.o
[ 46%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/states/trigger_state.cpp.o
[ 46%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/states/weaponskill_state.cpp.o
[ 46%] Building CXX object src/map/CMakeFiles/xi_map.dir/ai/ai_container.cpp.o
[ 47%] Building CXX object src/map/CMakeFiles/xi_map.dir/entities/automatonentity.cpp.o
[ 47%] Building CXX object src/map/CMakeFiles/xi_map.dir/entities/baseentity.cpp.o
[ 47%] Building CXX object src/map/CMakeFiles/xi_map.dir/entities/battleentity.cpp.o
[ 47%] Building CXX object src/map/CMakeFiles/xi_map.dir/entities/charentity.cpp.o
[ 47%] Building CXX object src/map/CMakeFiles/xi_map.dir/entities/mobentity.cpp.o
[ 48%] Building CXX object src/map/CMakeFiles/xi_map.dir/entities/npcentity.cpp.o
[ 48%] Building CXX object src/map/CMakeFiles/xi_map.dir/entities/petentity.cpp.o
[ 48%] Building CXX object src/map/CMakeFiles/xi_map.dir/entities/trustentity.cpp.o
[ 48%] Building CXX object src/map/CMakeFiles/xi_map.dir/items/item_currency.cpp.o
[ 49%] Building CXX object src/map/CMakeFiles/xi_map.dir/items/item_equipment.cpp.o
[ 49%] Building CXX object src/map/CMakeFiles/xi_map.dir/items/item_fish.cpp.o
[ 49%] Building CXX object src/map/CMakeFiles/xi_map.dir/items/item_flowerpot.cpp.o
[ 49%] Building CXX object src/map/CMakeFiles/xi_map.dir/items/item_furnishing.cpp.o
[ 49%] Building CXX object src/map/CMakeFiles/xi_map.dir/items/item_general.cpp.o
[ 50%] Building CXX object src/map/CMakeFiles/xi_map.dir/items/item_linkshell.cpp.o
[ 50%] Building CXX object src/map/CMakeFiles/xi_map.dir/items/item_puppet.cpp.o
[ 50%] Building CXX object src/map/CMakeFiles/xi_map.dir/items/item_shop.cpp.o
[ 50%] Building CXX object src/map/CMakeFiles/xi_map.dir/items/item_usable.cpp.o
[ 51%] Building CXX object src/map/CMakeFiles/xi_map.dir/items/item_weapon.cpp.o
[ 51%] Building CXX object src/map/CMakeFiles/xi_map.dir/items/item.cpp.o
[ 51%] Building CXX object src/map/CMakeFiles/xi_map.dir/los/los_tree_node.cpp.o
[ 51%] Building CXX object src/map/CMakeFiles/xi_map.dir/los/los_tree.cpp.o
[ 51%] Building CXX object src/map/CMakeFiles/xi_map.dir/los/zone_los.cpp.o
[ 52%] Building CXX object src/map/CMakeFiles/xi_map.dir/lua/lua_ability.cpp.o
[ 52%] Building CXX object src/map/CMakeFiles/xi_map.dir/lua/lua_action.cpp.o
[ 52%] Building CXX object src/map/CMakeFiles/xi_map.dir/lua/lua_baseentity.cpp.o
[ 52%] Building CXX object src/map/CMakeFiles/xi_map.dir/lua/lua_battlefield.cpp.o
[ 53%] Building CXX object src/map/CMakeFiles/xi_map.dir/lua/lua_instance.cpp.o
[ 53%] Building CXX object src/map/CMakeFiles/xi_map.dir/lua/lua_item.cpp.o
[ 53%] Building CXX object src/map/CMakeFiles/xi_map.dir/lua/lua_loot.cpp.o
[ 53%] Building CXX object src/map/CMakeFiles/xi_map.dir/lua/lua_mobskill.cpp.o
[ 54%] Building CXX object src/map/CMakeFiles/xi_map.dir/lua/lua_petskill.cpp.o
[ 54%] Building CXX object src/map/CMakeFiles/xi_map.dir/lua/lua_spell.cpp.o
[ 54%] Building CXX object src/map/CMakeFiles/xi_map.dir/lua/lua_statuseffect.cpp.o
[ 54%] Building CXX object src/map/CMakeFiles/xi_map.dir/lua/lua_trade_container.cpp.o
[ 54%] Building CXX object src/map/CMakeFiles/xi_map.dir/lua/lua_trigger_area.cpp.o
[ 55%] Building CXX object src/map/CMakeFiles/xi_map.dir/lua/lua_zone.cpp.o
[ 55%] Building CXX object src/map/CMakeFiles/xi_map.dir/lua/luautils.cpp.o
[ 55%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/action.cpp.o
[ 55%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/auction_house.cpp.o
[ 56%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/bazaar_check.cpp.o
[ 56%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/bazaar_close.cpp.o
[ 56%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/bazaar_confirmation.cpp.o
[ 56%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/bazaar_item.cpp.o
[ 56%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/bazaar_message.cpp.o
[ 57%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/bazaar_purchase.cpp.o
[ 57%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/blacklist.cpp.o
[ 57%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/campaign_map.cpp.o
[ 57%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/caught_fish.cpp.o
[ 58%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/caught_monster.cpp.o
[ 58%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/change_music.cpp.o
[ 58%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/char.cpp.o
[ 58%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/char_abilities.cpp.o
[ 58%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/char_appearance.cpp.o
[ 59%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/char_check.cpp.o
[ 59%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/char_emotion.cpp.o
[ 59%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/char_emotion_jump.cpp.o
[ 59%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/char_equip.cpp.o
[ 60%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/char_health.cpp.o
[ 60%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/char_job_extra.cpp.o
[ 60%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/char_jobs.cpp.o
[ 60%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/char_mounts.cpp.o
[ 60%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/char_recast.cpp.o
[ 61%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/char_skills.cpp.o
[ 61%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/char_spells.cpp.o
[ 61%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/char_stats.cpp.o
[ 61%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/char_sync.cpp.o
[ 62%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/char_update.cpp.o
[ 62%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/chat_message.cpp.o
[ 62%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/chocobo_digging.cpp.o
[ 62%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/conquest_map.cpp.o
[ 62%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/cs_position.cpp.o
[ 63%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/currency1.cpp.o
[ 63%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/currency2.cpp.o
[ 63%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/delivery_box.cpp.o
[ 63%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/downloading_data.cpp.o
[ 64%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/entity_animation.cpp.o
[ 64%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/entity_set_name.cpp.o
[ 64%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/entity_enable_list.cpp.o
[ 64%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/entity_update.cpp.o
[ 65%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/entity_visual.cpp.o
[ 65%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/event.cpp.o
[ 65%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/event_string.cpp.o
[ 65%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/event_update.cpp.o
[ 65%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/event_update_string.cpp.o
[ 66%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/fishing.cpp.o
[ 66%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/furniture_interact.cpp.o
[ 66%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/guild_menu.cpp.o
[ 66%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/guild_menu_buy.cpp.o
[ 67%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/guild_menu_buy_update.cpp.o
[ 67%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/guild_menu_sell.cpp.o
[ 67%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/guild_menu_sell_update.cpp.o
[ 67%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/independent_animation.cpp.o
[ 67%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/instance_entry.cpp.o
[ 68%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/inventory_assign.cpp.o
[ 68%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/inventory_count.cpp.o
[ 68%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/inventory_finish.cpp.o
[ 68%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/inventory_item.cpp.o
[ 69%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/inventory_modify.cpp.o
[ 69%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/inventory_size.cpp.o
[ 69%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/jobpoint_details.cpp.o
[ 69%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/jobpoint_update.cpp.o
[ 69%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/key_items.cpp.o
[ 70%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/linkshell_equip.cpp.o
[ 70%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/linkshell_message.cpp.o
[ 70%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/lock_on.cpp.o
[ 70%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/macroequipset.cpp.o
[ 71%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/map_marker.cpp.o
[ 71%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/menu_config.cpp.o
[ 71%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/menu_jobpoints.cpp.o
[ 71%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/menu_merit.cpp.o
[ 71%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/menu_mog.cpp.o
[ 72%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/menu_raisetractor.cpp.o
[ 72%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/merit_points_categories.cpp.o
[ 72%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/menu_unity.cpp.o
[ 72%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/message_basic.cpp.o
[ 73%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/message_combat.cpp.o
[ 73%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/message_name.cpp.o
[ 73%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/message_special.cpp.o
[ 73%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/message_standard.cpp.o
[ 74%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/message_system.cpp.o
[ 74%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/message_text.cpp.o
[ 74%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/monipulator1.cpp.o
[ 74%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/monipulator2.cpp.o
[ 74%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/party_define.cpp.o
[ 75%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/party_effects.cpp.o
[ 75%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/party_invite.cpp.o
[ 75%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/party_map.cpp.o
[ 75%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/party_member_update.cpp.o
[ 76%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/party_search.cpp.o
[ 76%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/pet_sync.cpp.o
[ 76%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/position.cpp.o
[ 76%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/quest_mission_log.cpp.o
[ 76%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/release.cpp.o
[ 77%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/release_special.cpp.o
[ 77%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/roe_questlog.cpp.o
[ 77%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/roe_sparkupdate.cpp.o
[ 77%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/roe_update.cpp.o
[ 78%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/server_ip.cpp.o
[ 78%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/server_message.cpp.o
[ 78%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/shop_appraise.cpp.o
[ 78%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/shop_buy.cpp.o
[ 78%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/shop_items.cpp.o
[ 79%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/shop_menu.cpp.o
[ 79%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/status_effects.cpp.o
[ 79%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/stop_downloading.cpp.o
[ 79%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/synth_animation.cpp.o
[ 80%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/synth_message.cpp.o
[ 80%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/synth_result.cpp.o
[ 80%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/synth_suggestion.cpp.o
[ 80%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/timer_bar_util.cpp.o
[ 80%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/trade_action.cpp.o
[ 81%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/trade_item.cpp.o
[ 81%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/trade_request.cpp.o
[ 81%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/trade_update.cpp.o
[ 81%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/treasure_find_item.cpp.o
[ 82%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/treasure_lot_item.cpp.o
[ 82%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/weather.cpp.o
[ 82%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/wide_scan.cpp.o
[ 82%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/wide_scan_track.cpp.o
[ 82%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/world_pass.cpp.o
[ 83%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/zone_in.cpp.o
[ 83%] Building CXX object src/map/CMakeFiles/xi_map.dir/packets/zone_visited.cpp.o
[ 83%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/attackutils.cpp.o
[ 83%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/battleutils.cpp.o
[ 84%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/blacklistutils.cpp.o
[ 84%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/blueutils.cpp.o
[ 84%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/charutils.cpp.o
[ 84%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/fishingutils.cpp.o
[ 85%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/gardenutils.cpp.o
[ 85%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/guildutils.cpp.o
[ 85%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/instanceutils.cpp.o
[ 85%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/itemutils.cpp.o
[ 85%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/jailutils.cpp.o
[ 86%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/mobutils.cpp.o
[ 86%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/moduleutils.cpp.o
[ 86%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/petutils.cpp.o
[ 86%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/puppetutils.cpp.o
[ 87%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/serverutils.cpp.o
[ 87%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/synthutils.cpp.o
[ 87%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/trustutils.cpp.o
[ 87%] Building CXX object src/map/CMakeFiles/xi_map.dir/utils/zoneutils.cpp.o
[ 87%] Building CXX object src/map/CMakeFiles/xi_map.dir/ability.cpp.o
[ 88%] Building CXX object src/map/CMakeFiles/xi_map.dir/alliance.cpp.o
[ 88%] Building CXX object src/map/CMakeFiles/xi_map.dir/anticheat.cpp.o
[ 88%] Building CXX object src/map/CMakeFiles/xi_map.dir/attack.cpp.o
[ 88%] Building CXX object src/map/CMakeFiles/xi_map.dir/attackround.cpp.o
[ 89%] Building CXX object src/map/CMakeFiles/xi_map.dir/autotranslate.cpp.o
[ 89%] Building CXX object src/map/CMakeFiles/xi_map.dir/battlefield_handler.cpp.o
[ 89%] Building CXX object src/map/CMakeFiles/xi_map.dir/battlefield.cpp.o
[ 89%] Building CXX object src/map/CMakeFiles/xi_map.dir/blue_spell.cpp.o
[ 89%] Building CXX object src/map/CMakeFiles/xi_map.dir/blue_trait.cpp.o
[ 90%] Building CXX object src/map/CMakeFiles/xi_map.dir/campaign_handler.cpp.o
[ 90%] Building CXX object src/map/CMakeFiles/xi_map.dir/campaign_system.cpp.o
[ 90%] Building CXX object src/map/CMakeFiles/xi_map.dir/char_recast_container.cpp.o
[ 90%] Building CXX object src/map/CMakeFiles/xi_map.dir/command_handler.cpp.o
[ 91%] Building CXX object src/map/CMakeFiles/xi_map.dir/conquest_system.cpp.o
[ 91%] Building CXX object src/map/CMakeFiles/xi_map.dir/daily_system.cpp.o
[ 91%] Building CXX object src/map/CMakeFiles/xi_map.dir/enmity_container.cpp.o
[ 91%] Building CXX object src/map/CMakeFiles/xi_map.dir/grades.cpp.o
[ 91%] Building CXX object src/map/CMakeFiles/xi_map.dir/guild.cpp.o
[ 92%] Building CXX object src/map/CMakeFiles/xi_map.dir/instance_loader.cpp.o
[ 92%] Building CXX object src/map/CMakeFiles/xi_map.dir/instance.cpp.o
[ 92%] Building CXX object src/map/CMakeFiles/xi_map.dir/item_container.cpp.o
[ 92%] Building CXX object src/map/CMakeFiles/xi_map.dir/job_points.cpp.o
[ 93%] Building CXX object src/map/CMakeFiles/xi_map.dir/latent_effect_container.cpp.o
[ 93%] Building CXX object src/map/CMakeFiles/xi_map.dir/latent_effect.cpp.o
[ 93%] Building CXX object src/map/CMakeFiles/xi_map.dir/linkshell.cpp.o
[ 93%] Building CXX object src/map/CMakeFiles/xi_map.dir/map.cpp.o
[ 94%] Building CXX object src/map/CMakeFiles/xi_map.dir/merit.cpp.o
[ 94%] Building CXX object src/map/CMakeFiles/xi_map.dir/message.cpp.o
[ 94%] Building CXX object src/map/CMakeFiles/xi_map.dir/mob_spell_container.cpp.o
[ 94%] Building CXX object src/map/CMakeFiles/xi_map.dir/mob_spell_list.cpp.o
[ 94%] Building CXX object src/map/CMakeFiles/xi_map.dir/mobskill.cpp.o
[ 95%] Building CXX object src/map/CMakeFiles/xi_map.dir/modifier.cpp.o
[ 95%] Building CXX object src/map/CMakeFiles/xi_map.dir/navmesh.cpp.o
[ 95%] Building CXX object src/map/CMakeFiles/xi_map.dir/notoriety_container.cpp.o
[ 95%] Building CXX object src/map/CMakeFiles/xi_map.dir/packet_guard.cpp.o
[ 96%] Building CXX object src/map/CMakeFiles/xi_map.dir/packet_system.cpp.o
[ 96%] Building CXX object src/map/CMakeFiles/xi_map.dir/party.cpp.o
[ 96%] Building CXX object src/map/CMakeFiles/xi_map.dir/petskill.cpp.o
[ 96%] Building CXX object src/map/CMakeFiles/xi_map.dir/recast_container.cpp.o
[ 96%] Building CXX object src/map/CMakeFiles/xi_map.dir/roe.cpp.o
[ 97%] Building CXX object src/map/CMakeFiles/xi_map.dir/spell.cpp.o
[ 97%] Building CXX object src/map/CMakeFiles/xi_map.dir/status_effect_container.cpp.o
[ 97%] Building CXX object src/map/CMakeFiles/xi_map.dir/status_effect.cpp.o
[ 97%] Building CXX object src/map/CMakeFiles/xi_map.dir/time_server.cpp.o
[ 98%] Building CXX object src/map/CMakeFiles/xi_map.dir/timetriggers.cpp.o
[ 98%] Building CXX object src/map/CMakeFiles/xi_map.dir/trade_container.cpp.o
[ 98%] Building CXX object src/map/CMakeFiles/xi_map.dir/trait.cpp.o
[ 98%] Building CXX object src/map/CMakeFiles/xi_map.dir/transport.cpp.o
[ 98%] Building CXX object src/map/CMakeFiles/xi_map.dir/treasure_pool.cpp.o
[ 99%] Building CXX object src/map/CMakeFiles/xi_map.dir/trigger_area.cpp.o
[ 99%] Building CXX object src/map/CMakeFiles/xi_map.dir/unitychat.cpp.o
[ 99%] Building CXX object src/map/CMakeFiles/xi_map.dir/universal_container.cpp.o
[ 99%] Building CXX object src/map/CMakeFiles/xi_map.dir/weapon_skill.cpp.o
[100%] Building CXX object src/map/CMakeFiles/xi_map.dir/zone_entities.cpp.o
[100%] Building CXX object src/map/CMakeFiles/xi_map.dir/zone_instance.cpp.o
[100%] Building CXX object src/map/CMakeFiles/xi_map.dir/zone.cpp.o
[100%] Linking CXX executable ../../../xi_map
[100%] Built target xi_map
make: *** [Makefile:171: all] Error 2
```

--- a/cmake/StandardProjectSettings.cmake
+++ b/cmake/StandardProjectSettings.cmake
@@ -111,3 +111,14 @@ function(disable_lto target)
     target_compile_options(${target} PRIVATE -fno-lto)
     target_link_options(${target} PRIVATE -fno-lto)
 endfunction()
+
+# If we're on Unix and the system is 32-bit (void* is 4-bytes wide),
+# then there's a good chance we're compiling for Raspberry Pi.
+# Currently, CMake doesn't detect this properly and needs some help
+# to link libatomic.
+# Source: https://gitlab.kitware.com/cmake/cmake/-/issues/21174
+#
+# TODO: Use include(CheckCXXSourceCompiles) to make this check better.
+if(UNIX AND CMAKE_SIZEOF_VOID_P EQUAL 4)
+    set(CMAKE_CXX_LINK_FLAGS "${CMAKE_CXX_LINK_FLAGS} -latomic")
+endif()

